lib/mayaUsd/ufe/ProxyShapeHierarchy.cpp

```diff
@@ -251,6 +251,18 @@ ProxyShapeHierarchy::insertChild(const Ufe::SceneItem::Ptr& child, const Ufe::Sc
     return insertChildCommand->insertedChild();
 }
 
+Ufe::InsertChildCommand::Ptr
+ProxyShapeHierarchy::appendChildVerifyRestrictionsCmd(const Ufe::SceneItem::Ptr& child)
+{
+    const auto childCmd = appendChildCmd(child);
+
+    if (childCmd && isConnected(downcast(child))) {
+        throw std::runtime_error("The node you're trying to move has connections.");
+    }
```

Comment:
In your Ufe PR it states "Returns a null pointer if the restrictions are not observed." - but here you are throwing. It should be one of the. Either return nullptr here or in Ufe change the doc to state that it will throw runtime_error if the restrictions are not observed.